python/pyspark/ml/torch/distributor.py

```diff
@@ -581,11 +593,11 @@ def _run_distributed_training(
             f"Started distributed training with {self.num_processes} executor proceses"
         )
         try:
+            assert self.spark is not None
             result = (
-                self.sc.parallelize(range(self.num_tasks), self.num_tasks)
-                .barrier()
-                .mapPartitions(spark_task_function)
-                .collect()[0]
+                self.spark.range(start=0, end=self.num_tasks, step=1, numPartitions=self.num_tasks)
+                .mapInPandas(func=spark_task_function, schema="output binary", barrier=True)
+                .first()["output"]
```

Comment:
I think barrier mode task does not support `.first()` operation if I remember it correctly

Reply:
got it, will fix